src/main/java/pathfinding/ConcreteGraph.java

```diff
@@ -111,17 +111,24 @@ public int compareTo(ASTNode node) {
 	{
 		int i;
 		ArrayList<Node> path = findPath(start, end);
+		int size = path.size();
+		Node hold;
+		for (int j = 0; j < size/2; j++)
```

Comment:
Don't declare that `size` variable, just do `for (int j = 0; j < path.size() / 2; j++)`. Do the same all the other uses of `size`.